datafusion/optimizer/src/eliminate_duplicated_expr.rs

```diff
@@ -41,15 +42,28 @@ impl OptimizerRule for EliminateDuplicatedExpr {
     ) -> Result<Option<LogicalPlan>> {
         match plan {
             LogicalPlan::Sort(sort) => {
+                let normalized_sort_keys = sort
+                    .expr
+                    .iter()
+                    .map(|e| match e {
+                        Expr::Sort(ExprSort { expr, .. }) => {
+                            Expr::Sort(ExprSort::new(expr.clone(), true, false))
```

Comment:
I am not sure about always using `true`, `fase`

Using the example
```sql
select * from t1 order by id desc, id, name, id asc;
select * from t1 order by id asc, id, name, id desc;
```

Is the sort key always the same ? It seems like in the first query the sort key should be `id desc, name` and the second it should be `id asc, name`

Maybe I am missing something

Reply:
Your conclusion is right.  The `true`, `fase` is just for normalize and dedup.

Reply:
Maybe we can add a test to cover this case as well?

Reply:
> Maybe we can add a test to cover this case as well?

Done.